ast35/Python/ast.c

```diff
@@ -1304,10 +1320,16 @@ handle_keywordonly_args(struct compiling *c, const node *n, int start,
                 if (!arg)
                     goto error;
                 asdl_seq_SET(kwonlyargs, j++, arg);
-                i += 2; /* the name and the comma */
+                i += 1 + (TYPE(CHILD(n, i + 1)) == COMMA); /* the name and the comma, if present */
```

Comment:
Yeah, that's probably slightly more readable.  Changed.

(It's not worth making the comma part of the switch because we have to do this check in the middle of varargs in 3.5, and it's worth maintaining parallel structure.  Also, in general I'd rather optimize for perf, like the surrounding code.)